indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -92,6 +92,28 @@ public static Path distributedClassPath(Path base)
     return new Path(base, "classpath");
   }
 
+  public static void authenticate(HadoopDruidIndexerConfig config)
+  {
+    String principal = config.getSchema().getTuningConfig().getPrincipal();
+    String keytab = config.getSchema().getTuningConfig().getKeytab();
+    if (Strings.isNullOrEmpty(principal) || Strings.isNullOrEmpty(keytab)) {
+      return;
+    }
+    Configuration conf = new Configuration();
+    UserGroupInformation.setConfiguration(conf);
+    if (UserGroupInformation.isSecurityEnabled()) {
+      try {
+        if (UserGroupInformation.getCurrentUser().hasKerberosCredentials() == false
+            || UserGroupInformation.getCurrentUser().getUserName() != principal) {
+          log.info("trying to authenticate user [%s] with keytab [%s]", principal, keytab);
+          UserGroupInformation.loginUserFromKeytab(principal, keytab);
```

Comment:
will anything bad happen if a user is logged in multiple times? (HDFS module might have already done it)